Spanner/tests/Snippet/TransactionTest.php

```diff
@@ -116,6 +118,56 @@ public function testExecuteUpdate()
         $this->assertEquals(1, $res->returnVal());
     }
 
+    public function testExecuteUpdateWithStruct()
+    {
+
+        $expectedSql = "UPDATE Posts SET title = 'Updated Title' WHERE " .
+                       "STRUCT<Title STRING, Content STRING>(Title, Content) = @post";
+
+        $expectedParams = [
+            'post' => ["Updated Title", "Sample Content"]
+        ];
+        $expectedStructData = [
+            "fields" => [
+                [
+                    "name" => "Title",
+                    "type" => [
+                        "code" => Database::TYPE_STRING
+                    ]
+                ],
+                [
+                    "name" => "Content",
+                    "type" => [
+                        "code" => Database::TYPE_STRING
+                    ]
+                ]
+            ]
+        ];
+
+        $this->connection->executeStreamingSql(Argument::allOf(
+            Argument::withEntry('sql', $expectedSql),
+            Argument::withEntry('params', $expectedParams),
+            Argument::withEntry(
+                'paramTypes',
+                Argument::withEntry('post', Argument::withEntry('structType', $expectedStructData))
```

Comment:
you can assert on nested array data? cool! TIL.

Reply:
Yeah! I found an example someone in phpspec, I'll send it to you if I can find it again.